apiserver/facades/client/applicationoffers/base.go

```diff
@@ -53,31 +55,27 @@ func (api *BaseAPI) checkAdmin(
 
 // checkControllerAdmin ensures that the logged in user is a controller admin.
 func (api *BaseAPI) checkControllerAdmin(ctx context.Context) error {
-	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, api.ControllerModel.ControllerTag())
+	controllerTag := names.NewControllerTag(api.controllerUUID)
+	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, controllerTag)
 }
 
-// modelForName looks up the model details for the named model and returns
-// the model (if found), the absolute model path which was used in the lookup,
-// and a bool indicating if the model was found,
-func (api *BaseAPI) modelForName(modelName, ownerName string) (Model, string, bool, error) {
+// modelForName returns the model details for the specified model name,
+// along with the absolute model path used in the lookup.
+//
+// The following errors may be returned:
+// - [modelerrors.NotFound] when no model with the given name exists.
```

Comment:
I think you want to remove this line? We explicitly throw away this error below so it can't be returned.

Reply:
I think see comments below before doing this.